src/document.js

```diff
@@ -18,8 +18,25 @@ const PostMessages = new PostMessageService({
 	loolframe: () => document.getElementById('loleafletframe').contentWindow
 })
 
-const showLoadingIndicator = () => document.getElementById('loadingContainer').classList.add('icon-loading')
-const hideLoadingIndicator = () => document.getElementById('loadingContainer').classList.remove('icon-loading')
+const showLoadingIndicator = () => {
+	if (OC.appswebroots.richdocumentscode && Config.get('urlsrc').indexOf('proxy.php') >= 0) {
+		var url = Config.get('urlsrc').substr(0, Config.get('urlsrc').indexOf('proxy.php') + 'proxy.php'.length)
+		$.get(url + '?status').done(function(val) {
+			if (val && val.status && val.status === 'starting') {
+				document.getElementById('proxyLoadingIcon').classList.add('icon-loading-small')
+				document.getElementById('proxyLoadingMessage').innerHTML = t('richdocuments', 'Built-in CODE Server is starting up shortly, please wait.')
```

Comment:
Let's use `textContent` instead of `innerHTML` otherwise translations which are synced from transifex could put random HTML in here.